visualization/app/codeCharta/ui/matchingFilesCounter/matchingFilesCounter.component.ts

```diff
@@ -0,0 +1,61 @@
+import "./matchingFilesCounter.component.scss"
+import { BlacklistType, BlacklistItem, CodeMapNode, Settings, RecursivePartial } from "../../codeCharta.model";
+import { SettingsServiceSubscriber, SettingsService } from "../../state/settings.service";
+import { CodeMapHelper } from "../../util/codeMapHelper";
+import { IRootScopeService, IAngularEvent } from "angular";
+import { NodeSearchService, NodeSearchSubscriber } from "../../state/nodeSearch.service";
+
+export class MatchingFilesCounterController implements NodeSearchSubscriber, SettingsServiceSubscriber {
+
+    private _viewModel: {
+		fileCount: number
+		hideCount: number
+		excludeCount: number
+		searchPattern: string
+		blacklist: BlacklistItem[]
+	} = {
+		fileCount: 0,
+		hideCount: 0,
+		excludeCount: 0,
+		searchPattern: "",
+		blacklist: []
+	}
+
+	private searchedNodeLeaves: CodeMapNode[] = [] // TODO: geht das in die DynamicSettings oder service
+
+	constructor($rootScope: IRootScopeService) {
+			NodeSearchService.subscribe($rootScope, this)
+			SettingsService.subscribe($rootScope, this)
+		}
+
+	public onNodeSearchComplete(searchedNodes: CodeMapNode[], event: angular.IAngularEvent){
+		this.searchedNodeLeaves = this.getSearchedNodeLeaves(searchedNodes)
+		this.updateViewModel(this.searchedNodeLeaves, this._viewModel.blacklist)
+	}
+
+	public onSettingsChanged(settings: Settings, update: RecursivePartial<Settings>, event: IAngularEvent){
+		this._viewModel.blacklist = settings.fileSettings.blacklist
```

Comment:
I think we should check if the blacklist is in the update object, as we might run this code too often, when its not necessary

Reply:
I don't think that this has an impact on performance, but i changed it anyways in order to be consistent with other components